postgre-client/src/asset_filter_client.rs

```diff
@@ -137,18 +162,29 @@ impl PgClient {
                                 " AND assets_v3.ast_is_collection_verified IS DISTINCT FROM FALSE",
                             );
                         }
+
+                        query_builder.push(" ORDER BY ");
+                        query_builder.push(order.sort_by.to_string());
+                        query_builder.push(direction);
+
+                        query_builder.push(" LIMIT ");
+                        if let Some(page_num) = page.filter(|&p| p > 1) {
+                            let lim = (page_num - 1) * limit;
```

Comment:
The condition is wrong. it's page_num*limit. You want to have 2000 records pre filtered for the second page. Also add a test for the second page of pagination of this request. The correct way to do it is you always get a 1-based page and multiply it by the limit